ngraph/core/src/function.cpp

```diff
@@ -266,7 +283,20 @@ std::vector<shared_ptr<ov::Node>> ov::Function::get_ordered_ops() const {
         nodes.push_back(param);
     }
 
-    return m_topological_sorter(nodes);
+    auto order = m_topological_sorter(nodes);
+
+    const lock_guard<mutex> lock(m_topological_sort_mutex);
+
+    // Update nodes cache and update all nodes to have shared rt info
+    // which belongs to the current Function.
+    m_cached_ordered_ops.clear();
+    for_each(order.cbegin(), order.cend(), [this](shared_ptr<Node> node) {
+        m_cached_ordered_ops.push_back(node);
+        node->m_shared_rt_info.insert(m_shared_rt_info);
```

Comment:
Am I right that this approach doesn't work if node is shared between several functions?

Reply:
No, Node stores set of shared objects which belong to different functions. So we support cases when node is included into multiple Functions.